### PR TITLE
Plumb contexts through commands

### PR DESCRIPTION
**Pull request by thaJeztah**
- relates to / carries first commit from https://github.com/docker/cli/pull/4556
- relates to https://github.com/docker/cli/pull/4698

This is to prepare for otel support.

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: cli/command/container/stats.go [hunk 1]
```diff
@@ -128,13 +127,16 @@ func runStats(dockerCli command.Cli, opts *statsOptions) error {
 	}
 
 	if showAll {
+		span := trace.SpanFromContext(ctx)
+
```
Comment (thaJeztah) [inline cli/command/container/stats.go:131]: DOH! Looks like some OTEL changes landed in the wrong commit; let me fix that;

```
13 7.030 ERROR: Vendor result differs. Please vendor your package with "make -f docker.Makefile vendor"
```

File: cli/command/cli.go [hunk 2]
```diff
@@ -82,6 +82,16 @@ type DockerCli struct {
...
+	// baseCtx is the base context used for internal operations
+	// In the future this may be replaced by explicitly passing a context to functions that need it.
+	baseCtx context.Context
+}
+
+// WithContext sets the base context for the cli
+// This is used internally for operations may require a context to propagate tracing.
+func (cli *DockerCli) WithContext(ctx context.Context) {
+	cli.baseCtx = ctx
 }
```
Comment (thaJeztah) [inline cli/command/cli.go:96]: I'm moving this one to the OTEL commit; it's not used yet in this commit, and I want to look if this should be a `DockerCliOption` instead (also see https://github.com/docker/cli/pull/4387)

**Review by sam-thibault — APPROVED** [review]
LGTM

**Review by vvoland — APPROVED** [review]

File: cli/command/cli.go [hunk 3]
```diff
@@ -82,6 +82,11 @@ type DockerCli struct {
 	dockerEndpoint     docker.Endpoint
 	contextStoreConfig store.Config
 	initTimeout        time.Duration
+
+	// baseCtx is the base context used for internal operations. In the future
+	// this may be replaced by explicitly passing a context to functions that
+	// need it.
+	baseCtx context.Context
```
Comment (laurazard) [inline cli/command/cli.go:89]: Should keep an eye on this and see if we can factor it out later 👀

**Review by laurazard — APPROVED** [review]
LGTM

File: cli/command/cli.go (hunk 3 above)
Comment (thaJeztah) [reply cli/command/cli.go:89]: cc @cpuguy83

**Pull request merged**